dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/Provisioner.java

```diff
@@ -608,7 +588,9 @@ public boolean handle(FeatureResource fr) {
         for (Bundle bundleToUninstall : bundlesToUninstall) {
             try {
                 bundleToUninstall.uninstall();
+                installStatus.addBundleRemovedDelta(bundleToUninstall);
             } catch (IllegalStateException e) {
+                e.printStackTrace();
```

Comment:
Did you intend to do a printStackTrace here?

Reply:
It would seem that you should add to the status the intention to uninstall this bundle before calling uninstall().  The BundleProcessor no longer will get an uninstalled event so it would miss it if we somehow had another thread uninstall the bundle before we could here.

Reply:
Changed so the installstatus method is called before the uninstall